### DISCUSSION TITLE
display "national_park" in OpenGL

### DISCUSSION THREAD

**Discussion by cricri-du-lauragais** [osmandapp/OsmAnd#14997] [General]
### Description
display "national_park" when changing the color and zoom in a render.style child of the default style between OpenGL mode via the development plugin on Android and the old engine
44.11888° N, 3.58444° E

here the old engine (it's OK)
![old_engine_OsmAnd](https://user-images.githubusercontent.com/83398215/182031326-0ea16099-b014-4f13-bef5-dacebbe79f9f.jpg)

and OpenGL (only inner)
![opengl_OsmAnd](https://user-images.githubusercontent.com/83398215/182031336-00d48770-0eb9-4b33-aed9-9d2ed49e9b18.jpg)

### How to reproduce?
try this style 

```
<?xml version="1.0" encoding="utf-8"?>
<renderingStyle name="test" depends="default" defaultColor="#ebe7e4" version="1">

	<order>
	</order>

	<text>
	</text>

	<point>
	</point>
	
	<polygon>
			<switch>
				<switch natureReserves="true">
					<case minzoom="6" tag="leisure" value="nature_reserve" color_2="#4400ae3d"/>
					<case minzoom="8" tag="boundary" value="national_park" color_2="#4400ae3d"/>
					<apply color="#99ff0000" strokeWidth_2="2:2"/>
				</switch>
			</switch>
	</polygon>
	
	<line>
	
		<switch natureReserves="true">
			<case tag="boundary" value="national_park" minzoom="7">
				<case maxzoom="9" strokeWidth="5:5"/>
				<case minzoom="10" strokeWidth="6:6"/>
				<apply color="#ff0000" color_0="#0099ff" strokeWidth_0="2" pathEffect="4_1" pathEffect_0="4_1"/>
			</case>
			<case minzoom="8" tag="boundary" value="protected_area" color="#4400ae3d">
				<apply>
					<case maxzoom="13" strokeWidth="2:2" pathEffect="10_5"/>
					<case minzoom="14" strokeWidth="3.5:3.5" pathEffect="13_8"/>
				</apply>
			</case>
		</switch>


		<case tag="osmand_change" value="delete"/> <!-- OSM Live will break if delete -->
	</line>
</renderingStyle>
```
### Your Environment
OsmAnd Version: 4.2.7
Android/iOS version: 12
Device model: samsung galaxy A33 5G

**Maps used (online or offline):**   
offline vector map with live update

**Comment by cricri-du-lauragais** [comment]
it's the same with the default.render but it's hard to seeing :
![Screenshot_20220731-181723_OsmAnd](https://user-images.githubusercontent.com/83398215/182036104-3861cfae-ecb3-4769-ae82-5bbef1fc6472.jpg)
)

**Comment by cricri-du-lauragais** [comment]
I think it's a bug in the OpenGL engine ;)
not a discussion about rendering.

**Reply by vshcherb** [reply]
I see only 2 relations
https://www.openstreetmap.org/relation/1178041#map=15/44.1101/3.5902
https://www.openstreetmap.org/relation/1178102#map=13/44.1283/3.6037

So probably bug is about relation is not rendered in OpenGL please provided it and a small map.

**Comment by cricri-du-lauragais** [comment]
probably but the old engine handles it correctly (with nature_reserve when national_park does not work because a national park is necessarily a nature reserve)
